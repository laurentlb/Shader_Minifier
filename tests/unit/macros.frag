// Define a bunch of macros. Make sure Shader Minifier doesn't use
// these names when renaming variables.
// Also, test that the spaces are stripped.
#define a $
#define b (    $+)
#define c   $  +
#define d abc    def  + ghi ()
#define e $
#define f $
#define g $
#define h $
#define i $
#define j $
#define k $
#define l $
#define m $
#define n $
#define o $
#define p $

int foo() {
  int var1 = 1, var2 = 2, var3 = 3, var4 = 4, var5 = 5, var6 = 6;
  int var7 = 7, var8 = 8, var9 = 9, var10 = 10, var11 = 11, var12 = 12;
  return var1 + var5 + var12;
}
