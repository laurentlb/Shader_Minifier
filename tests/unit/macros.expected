// Generated with  (https://github.com/laurentlb/Shader_Minifier/)
#ifndef MACROS_EXPECTED_
# define MACROS_EXPECTED_

const char *macros_frag =
 "#define a$\n"
 "#define b ($+)\n"
 "#define c$+\n"
 "#define d abc def+ghi ()\n"
 "#define e$\n"
 "#define f$\n"
 "#define g$\n"
 "#define h$\n"
 "#define i$\n"
 "#define j$\n"
 "#define k$\n"
 "#define l$\n"
 "#define m$\n"
 "#define n$\n"
 "#define o$\n"
 "#define p$\n"
 "int E()"
 "{"
   "int E=1,r=2,C=3,B=4;"
   "C=5;"
   "r=12;"
   "return E+C+r;"
 "}";

#endif // MACROS_EXPECTED_
